### PR TITLE
Exchange idsite token

### PR DESCRIPTION
**Pull request by nikone**

### CODE REVIEW

**Comment by nikone** [comment]
Exchange ID Site token

```ruby
refresh_grant_request  = Stormpath::Oauth::IdSiteGrantRequest.new(jwt_token)
application.authenticate_oauth(refresh_grant_request) #=> returnes Stormpath::Resource::AccessToken
```

File: spec/resource/application_spec.rb [hunk 1]
```diff
@@ -654,6 +654,19 @@ def create_organization_account_store_mapping(organization, account_store)
       end
     end
 
+    context 'exhange id site token for access_token' do
```
Comment (ecrisostomo) [inline spec/resource/application_spec.rb]: Looks like this test is only for asserting failures. Can we get tests for verifying the happy paths?

**Comment by ecrisostomo** [comment]
Please add a small piece of documentation to the Readme.md file with the specs of how this feature works.

**Comment by nikone** [comment]
Added description to Readme

File: README.md [hunk 2]
```diff
@@ -340,6 +340,12 @@ Again, with all these methods, You will want your application to link to an inte
 > NOTE:
 > A JWT will expire after 60 seconds of creation.
 
+#### Exchange ID Site token for a Stormpath Access Token
+After the user has been authenticated via ID Site, a developer may want to control their authorization with an OAuth 2.0 Token. 
+This is done by passing the JWT similar to the way we passed the user’s credentials as described in [Generating an OAuth 2.0 Access Token][generate-oauth-access-token]. 
+The difference is that instead of using the password grant type and passing credentials, we will use the id_site_token type and pass the JWT we got from the ID Site
+more info [here][exchange-id-site-token].
```
Comment (ecrisostomo) [inline README.md]: I think we will need an example of how this works in Ruby. Just how the calls would look like for the user.

**Pull request closed without merging**